katsdpcontroller/product_controller.py

```diff
@@ -907,6 +899,43 @@ async def gain_impl(
         """
         raise NotImplementedError()
 
+    async def gain_all(self, stream_name: str, values: Sequence[str]) -> None:
+        """Set F-engine gains for all inputs."""
+        if self.state not in {ProductState.CAPTURING, ProductState.IDLE}:
+            raise FailReply(f"Cannot set gains in state {self.state}")
+        stream = self._find_stream(stream_name)
+        if not isinstance(stream, product_config.GpucbfAntennaChannelisedVoltageStream):
+            raise FailReply(f"Stream {stream_name!r} is of the wrong type")
+        if len(values) not in {1, stream.n_chans}:
+            raise FailReply(f"Expected 0, 1, or {stream.n_chans} values, received {len(values)}")
```

Comment:
Maybe I'm missing something, but your text says you expect possibly 0, but you don't check for that in the if-statement.

Reply:
Error in the comment - thanks (copy-pasted from `?gain`, which does support passing 0 values to query).